src/test/ProposalPayloadE2E.t.sol

```diff
@@ -3,37 +3,335 @@ pragma solidity ^0.8.15;
 
 // testing libraries
 import "@forge-std/Test.sol";
+import "@forge-std/console.sol";
 
 // contract dependencies
 import {GovHelpers} from "@aave-helpers/GovHelpers.sol";
-import {AaveV2Ethereum} from "@aave-address-book/AaveV2Ethereum.sol";
+import {BridgeExecutorHelpers} from "@aave-helpers/BridgeExecutorHelpers.sol";
+import {AaveV3Polygon} from "@aave-address-book/AaveV3Polygon.sol";
 import {ProposalPayload} from "../ProposalPayload.sol";
-import {DeployMainnetProposal} from "../../script/DeployMainnetProposal.s.sol";
+import {ProposalPayloadPolygon} from "../ProposalPayloadPolygon.sol";
+import {DeployProposals} from "../../script/DeployProposals.s.sol";
+import {ProtocolV3TestBase, InterestStrategyValues, ReserveConfig, ReserveTokens, IERC20} from "@aave-helpers/ProtocolV3TestBase.sol";
+import {IStateReceiver} from "@governance-crosschain-bridges/dependencies/polygon/fxportal/FxChild.sol";
+import {IDefaultInterestRateStrategy} from "@aave-v3-core/interfaces/IDefaultInterestRateStrategy.sol";
+import {AaveV2Helpers, InterestStrategyValues as InterestStrategyValuesV2, IReserveInterestRateStrategy as IReserveInterestRateStrategyV2, ReserveConfig as ReserveConfigV2} from "./utils/AaveV2Helpers.sol";
+import {AaveAddressBookV2} from "./utils/AaveAddressBookV2.sol";
 
-contract ProposalPayloadE2ETest is Test {
+contract ProposalPayloadE2ETest is ProtocolV3TestBase {
+    uint256 internal constant RAY = 1e27;
+    uint256 public mainnetFork;
+    uint256 public polygonFork;
     uint256 public proposalId;
+    ProposalPayloadPolygon public proposalPayloadPolygon;
+    ProposalPayload public proposalPayload;
+
+    // Old Strategies
+    IDefaultInterestRateStrategy public constant OLD_INTEREST_RATE_STRATEGY_V3_POLYGON =
+        IDefaultInterestRateStrategy(0x03733F4E008d36f2e37F0080fF1c8DF756622E6F);
+    IReserveInterestRateStrategyV2 public constant OLD_INTEREST_RATE_STRATEGY_POLYGON =
+        IReserveInterestRateStrategyV2(0x9025C2d672afA29f43cB59b3035CaCfC401F5D62);
+    IReserveInterestRateStrategyV2 public constant OLD_INTEREST_RATE_STRATEGY_ETHEREUM =
+        IReserveInterestRateStrategyV2(0xfC0Eace19AA7498e0f36eF1607D282a8d6debbDd);
+
+    // New Strategies
+    address public constant NEW_INTEREST_RATE_STRATEGY_ETHEREUM = 0x04c28D6fE897859153eA753f986cc249Bf064f71;
+    address public constant NEW_INTEREST_RATE_STRATEGY_POLYGON = 0x80cb7e9E015C5331bF34e06de62443d070FD6654;
+    address public constant NEW_INTEREST_RATE_STRATEGY_POLYGON_V3 = 0x4b8D3277d49E114C8F2D6E0B2eD310e29226fe16;
```

Comment:
Would recommend changing these to `immutable` and setting these from the respective proposal payloads in the setUp() function. That way it's not human error prone while copy pasting.